src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -131,10 +134,19 @@ public void runAppCommand(List<String> args) throws ProcessRunnerException {
    * Uses the process runner to execute a dev_appserver.py command.
    *
    * @param args the arguments to pass to dev_appserver.py
-   * @throws InvalidPathException when Python can't be located
-   * @throws ProcessRunnerException when process runner encounters an error
+   * @throws InvalidPathException      when Python can't be located
+   * @throws ProcessRunnerException    when process runner encounters an error
+   * @throws CloudSdkNotFoundException when the Cloud SDK is not installed where expected
+   * @throws AppEngineException        when dev_appserver.py cannot be found
    */
   public void runDevAppServerCommand(List<String> args) throws ProcessRunnerException {
+    validateCloudSdk();
+    if (!Files.isRegularFile(getDevAppServerPath())) {
```

Comment:
the dev-appserver is always included with the cloud sdk, the
app-engine-java component is not required for running the dev-appserver,
but apparently you need to invoke it correctly so that it'll auto-install
the java component when it detects a java runtime is needed.

On Mon, Sep 12, 2016 at 3:45 PM, meltsufin <notifications@github.com> wrote:

> In src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java
> <https://github.com/GoogleCloudPlatform/appengine-plugins-core/pull/209#discussion_r78441010>
> :
>
> >     */
> >    public void runDevAppServerCommand(List<String> args) throws ProcessRunnerException {
> > +    validateCloudSdk();
> > +    if (!Files.isRegularFile(getDevAppServerPath())) {
>
> Ss the app-engine-java component necessary and sufficient for running Java
> apps with dev_appserver.py?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/GoogleCloudPlatform/appengine-plugins-core/pull/209/files/b4db3945ca1769ffd1bde02f454625c91b3d3e57#r78441010>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHf5HSmW7CsZsSkMTnkqh9QirlmkGqloks5qpavagaJpZM4J68gI>
> .
>